modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+<?php
+class Jetpack_Widget_Social_Icons extends WP_Widget {
+	/**
+	 * @var array Default widget options.
+	 */
+	protected $defaults;
+
+	/**
+	 * @var array protocols that are allowed in esc_url validation function.
+	 */
+	protected $protocols = array( 'http', 'https', 'mailto', 'skype' );
+
+	/**
+	 * Widget constructor.
+	 */
+	public function __construct() {
+		$widget_ops = array(
+			'classname'                   => 'jetpack_widget_social_icons',
+			'description'                 => esc_html__( 'Add social-media icons to your site.', 'jetpack' ),
+			'customize_selective_refresh' => true,
+		);
+
+		parent::__construct( 'jetpack_widget_social_icons', esc_html__( 'Social Icons', 'jetpack' ), $widget_ops );
+
+		$this->defaults = array(
+			'title'     => esc_html__( 'Follow Us', 'jetpack' ),
+			'icon-size' => 'medium',
+			'icons'     => array(
+				array(
+					'url'   => '',
+				),
+			),
+		);
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );
+		add_action( 'admin_print_footer_scripts', array( $this, 'render_admin_js' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_icon_scripts' ) );
+		add_action( 'wp_footer', array( $this, 'jetpack_social_menu_include_svg_icons' ), 9999 );
```

Comment:
This looks bad.

Wouldn't this result in the SVGs printing on every `wp_footer` regardless of whether or not the widget is ever actually used, let alone used on that page (due to either multiple sidebars or widget visibility)?